middleware/app/Models/ProviderPlatform.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProviderPlatform extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'id',
+        'type',
+        'name',
+        'account_id',
+        'access_key',
+        'base_url',
+        'icon_url',
+    ];
+
+    public function __construct(array $attributes = [])
+    {
+        parent::__construct($attributes);
+    }
+
+    public static function getAllProviderIds(): array
```

Comment:
I'm not sure this is needed, but we can look at the ORM.

Reply:
I think something like this would work instead of needing this method: ProviderPlatform::select('id')->get();